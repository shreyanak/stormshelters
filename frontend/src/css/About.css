.about-container {
  background-color: #1a1a1a;
  color: #fff;
  padding: 20px;
  overflow-x: auto;
}

.about-card {
  width: 15%;
  padding: 20px;
  background-color: #333;
  border-radius: 10px;
  box-shadow: 0 4px 6px rgba(255, 255, 255, 0.1);
  transition: transform 0.2s, box-shadow 0.2s;
  margin: 20px;
  text-align: center;
}

.about-card:hover {
  transform: translateY(-5px);
  box-shadow: 0 8px 12px rgba(255, 255, 255, 0.2);
}

.team-members-container {
  display: flex;
  justify-content: flex-start;
  flex-wrap: nowrap;
  margin-top: 20px;
  padding-bottom: 20px;
  overflow-x: auto;
}

.center-text {
  text-align: center;
}

.statistics {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-start;
  margin-top: 20px;
}

.about-container p {
  color: #ccc;
}

.about-container h1,
.about-container h2 {
  color: #fff;
}

.about-container ul {
  list-style-type: none;
  padding: 0;
}

.about-container ul li {
  margin-bottom: 10px;
}

.about-card img {
  width: 150px;
  height: 150px;
  border-radius: 50%;
  object-fit: cover;
}