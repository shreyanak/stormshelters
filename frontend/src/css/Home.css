.home-container {
    position: relative;
    height: 100vh;
    overflow: hidden;
  }
  
  .bg {
    background-image: url("../img/splash.jpeg");
  
    height: 100%; 
  
    /* Center and scale */
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
  }
  