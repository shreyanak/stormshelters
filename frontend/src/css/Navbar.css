.navbar {
  background-color: #1a1a1a;
  box-shadow: 0px 2px 4px rgba(255, 255, 255, 0.2); /* Shadow effect */
  padding: 10px;
  color: #fff; 
}

.nav-list {
  list-style: none;
  padding: 0;
  display: flex;
  justify-content: space-around;
}

.nav-item {
  margin: 0;
}

.nav-link {
  text-decoration: none;
  color: #ffffff;
  font-weight: bold;
  padding: 10px;
  transition: transform 0.2s, box-shadow 0.2s;
}

/* Hover effect */
.nav-link:hover {
  transform: translateY(-2px);
  box-shadow: 0px 4px 6px rgba(255, 255, 255, 0.2);
}
