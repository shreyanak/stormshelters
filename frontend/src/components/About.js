import React, { useEffect, useState } from 'react';
import '../css/About.css';

function About() {
  const [contributorData, setContributorData] = useState([]);
  const [totalCommits, setTotalCommits] = useState(0);
  const [totalIssues, setTotalIssues] = useState(0);

  useEffect(() => {
    const projectId = '50438097';
    const contributorData = [
      { name: 'aaronk2711', displayName: 'Aaron Kulkarni' },
      { name: 'shreya.n', displayName: 'Shreya Nakka' },
      { name: 'rohitc28', displayName: 'Rohit Chawla' },
      { name: 'johnrsmith2003', displayName: 'John Smith' },
      { name: 'areye2020', displayName: 'Adriana Reyes' },
    ];

    async function fetchData() {
      let totalCommitsCount = 0;
      let totalIssuesCount = 0;

      for (const contributor of contributorData) {
        const { name, displayName } = contributor;

        // Fetch issues
        const issuesResponse = await fetch(`https://gitlab.com/api/v4/projects/${projectId}/issues?author_username=${name}`);
        const issuesData = await issuesResponse.json();
        const issuesCount = issuesData.length;

        // Fetch commits
        const commitsResponse = await fetch(`https://gitlab.com/api/v4/projects/${projectId}/repository/commits?author_username=${name}&per_page=100`);
        const commitsData = await commitsResponse.json();
        let commitsCount = 0;

        for (const commit of commitsData) {
          if (commit.author_name === displayName) {
            commitsCount++;
          }
        }

        totalCommitsCount += commitsCount;
        totalIssuesCount += issuesCount;

        setContributorData((prevData) => [
          ...prevData,
          {
            name: displayName,
            issues: issuesCount,
            commits: commitsCount,
          },
        ]);
      }

      setTotalCommits(totalCommitsCount);
      setTotalIssues(totalIssuesCount);
    }

    fetchData();
  }, []);

  return (
    <div className="about-container">
      <h1>About Page</h1>
      <p>
        StormShelters is an application that can provide those in Harris County
        that have been affected by natural disasters with resources for help and
        shelter. The website provides information on the different cities in the
        county, the disasters they were struck by, and different shelter locations
        nearby. Data sources used: Yelp, FEMA, Wikipedia, Census. Tools used: GitLab
        repository with CI/CD, Visual Studio Code, JavaScript, HTML/CSS, Postman,
        Bootstrap Through our research, we have found interesting correlations
        between data we didn't realize was so closely connected. The most
        interesting realization was how shelters were concentrated near the areas
        where many storms take place.
      </p>
      <div className="team-members-container">
        {contributorData.map((contributor, index) => (
          <div className="team-member-card" key={index}>
            <img src={`./static/img/${contributor.name.toLowerCase()}.jpg`} alt={`contributor ${index + 1}`} />
            <div className="text">{contributor.name}</div>
            <div className="text">{`${contributor.description}`}</div>
            <div className="text">{`Number of issues: ${contributor.issues}`}</div>
            <div className="text">{`Number of commits: ${contributor.commits}`}</div>
            <div className="text">Number of unit tests: 0</div>
          </div>
        ))}
      </div>
      <div className="statistics" style={{ justifyContent: 'left' }}>
        <p>{`Total commits: ${totalCommits}`}</p>
        <p>{`Total issues: ${totalIssues}`}</p>
      </div>
      <div className="center-text">
        <br />
        <a href="https://documenter.getpostman.com/view/29974721/2s9YJZ3jac">API Documentation</a>
        <br />
        <a href="https://gitlab.com/cs373-group5/stormshelters">GitLab Repository</a>
      </div>
      <div className="text-left">
        <br />
        <h2>Tools Used:</h2>
        <ul>
          <li>VS Code: integrated environment for command line and pushing to Git</li>
          <li>VS Code Liveshare: remotely edit the same project simultaneously</li>
          <li>Zoom: for remote collaboration</li>
          <li>AWS Amplify and Route53: Hosting the website directly from the GitLab repository and configuring the different domain provisions (i.e .dev, .amplify, .www)</li>
        </ul>

        <h2>Data sources:</h2>
        <ul>
          <li><a href="https://census.gov">Census</a></li>
          <li><a href="https://fusion.yelp.com">Yelp Fusion</a></li>
          <li><a href="https://www.mediawiki.org/wiki/API:Main_page">MediaWiki</a></li>
          <li><a href="https://www.fema.gov/about/reports-and-data/openfema">OpenFEMA</a></li>
        </ul>
        <p>Each of these sites provides a RESTful API. For Phase 1, we first analyzed the documentation to find the appropriate endpoints. Then, we scraped the data using a curl request and filtered the data by hand to find the appropriate information. Finally, we added the data to a JSON file so that it could be re-used across different pages.</p>
        <br />
      </div>
    </div>
  );
}

export default About;
