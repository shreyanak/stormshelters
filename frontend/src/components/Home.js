import React from 'react';
import Carousel from 'react-bootstrap/Carousel';
import { Link } from 'react-router-dom';
import '../css/Home.css';

function Home() {
  
  return (
    <div className="home-container">
      <div className="shelter-card-container">
        {/* Card for Cities */}
        <div className="shelter-card">
          <div className="shelter-card-body">
            <h3 className="shelter-card-title">Cities</h3>
            <p>
              View the weather conditions of your city to see if a storm is in the forecast. Please work.
            </p>
          </div>
          <div className="button-container">
          </div>
        </div>

        {/* Card for Shelters */}
        <div className="shelter-card">

          <div className="shelter-card-body">
            <h3 className="shelter-card-title">Shelters</h3>
            <p>
              View the areas to take refuge in the Houston area. Filter by address to find the most convenient location for you.
            </p>
          </div>
          <div className="button-container">
          </div>
        </div>

        {/* Card for Food and Pharmacies */}
        <div className="shelter-card">
          <div className="shelter-card-body">
            <h3 className="shelter-card-title">Food and Pharmacies</h3>
            <p>
              View the highest-rated locations where you can stock up for the next storm or replenish your supplies while you recover. Also find where you can purchase essential medical supplies closest to you.
            </p>
          </div>
        </div>
      </div>
      
    <Carousel>
      <Carousel.Item>Test</Carousel.Item>
      <Carousel.Item>Test</Carousel.Item>

    </Carousel>

    </div>

  );
}

export default Home;
