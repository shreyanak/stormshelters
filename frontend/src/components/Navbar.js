import React from 'react';
import '../css/Navbar.css';

function Navbar() {
  return (
    <nav className="navbar">
      <ul className="nav-list">
      <li className="nav-item">
          <a href="/" className="nav-link">
            StormShelters
          </a>
        </li>
        <li className="nav-item">
          <a href="/about" className="nav-link">
            About
          </a>
        </li>
        <li className="nav-item">
          <a href="/cities" className="nav-link">
            Cities
          </a>
        </li>
        <li className="nav-item">
          <a href="/shelter" className="nav-link">
            Shelter & Relief
          </a>
        </li>
        <li className="nav-item">
          <a href="/pharmacies" className="nav-link">
            Pharmacies
          </a>
        </li>
        <li className="nav-item">
          <a href="/our-visualizations" className="nav-link">
            Visualizations
          </a>
        </li>
        <li className="nav-item">
          <a href="/provider-visualizations" className="nav-link">
            Provider Visualizations
          </a>
        </li>
      </ul>
    </nav>
  );
}

export default Navbar;
