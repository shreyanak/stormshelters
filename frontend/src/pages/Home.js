import React from 'react';
import './Home.css';

function Home() {
  return (
    <div className="home-container">
      <div className="bg">
        {/* container for splash */}
      </div>
    </div>
  );
}

export default Home;
