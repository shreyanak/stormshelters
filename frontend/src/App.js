import React from 'react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import Navbar from './components/Navbar';
import Home from './components/Home';
import About from './components/About';
import Disasters from './components/Disasters';
import Shelters from './components/Shelter';
import Cities from './components/Cities';
import CityDetail from './components/CityDetail';
import ShelterDetail from './components/ShelterDetail';

function App() {
  return (
    <Router>
      <Navbar />
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/about" element={<About />} />
        <Route path="/cities" element={<Cities />} />
        <Route path="/disasters" element={<Disasters />} />
        <Route path="/shelter" element={<Shelters />} />
        <Route path="/city/:cityName" element={<CityDetail />} />
        <Route path="/shelter/:shelterName" element={<ShelterDetail />} />
      </Routes>
    </Router>
  );
}

export default App;
