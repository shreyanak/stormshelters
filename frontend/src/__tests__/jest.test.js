import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';


import App from '../App.js'

import Navbar from '../components/Navbar.js';

import About from '../components/About.js';
import Home from '../components/Home.js';
import CityCard from '../components/CityModel.js';

it('Init NavBar', () => {
    const component = renderer.create(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
  
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  

  it('Init About', () => {
    const component = renderer.create(
      <BrowserRouter>
        <About />
      </BrowserRouter>,
    );
  
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Init App', () => {
    const component = renderer.create(<App />);
  
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('Init HomeCard', () => {
    const component = renderer.create(
      <BrowserRouter>
        <Home />
      </BrowserRouter>,
    );
  
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

//   it('Init CityCard', () => {
//     const component = renderer.create(
//       <BrowserRouter>
//         <CityCard />
//       </BrowserRouter>,
//     );
  
//     let tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
//   });
  
  
  