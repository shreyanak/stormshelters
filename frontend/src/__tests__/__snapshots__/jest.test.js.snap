// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Init About 1`] = `
<div
  className="about-container"
>
  <h1>
    About Page
  </h1>
  <p>
    StormShelters is an application that can provide those in Harris County that have been affected by natural disasters with resources for help and shelter. The website provides information on the different cities in the county, the disasters they were struck by, and different shelter locations nearby. Data sources used: Yelp, FEMA, Wikipedia, Census. Tools used: GitLab repository with CI/CD, Visual Studio Code, JavaScript, HTML/CSS, Postman, Bootstrap Through our research, we have found interesting correlations between data we didn't realize was so closely connected. The most interesting realization was how shelters were concentrated near the areas where many storms take place.
  </p>
  <div
    className="team-members-container"
  />
  <div
    className="statistics"
  >
    <p>
      Total commits: 0
    </p>
    <p>
      Total issues: 0
    </p>
  </div>
  <div
    className="center-text"
  >
    <br />
    <a
      href="https://documenter.getpostman.com/view/29974721/2s9YR6aDwG"
    >
      API Documentation
    </a>
    <br />
    <a
      href="https://gitlab.com/cs373-group5/stormshelters"
    >
      GitLab Repository
    </a>
  </div>
  <div
    className="text-left"
  >
    <br />
    <h2>
      Tools Used:
    </h2>
    <ul>
      <li>
        VS Code: Integrated Development Environment
      </li>
      <li>
        Zoom: Remote collaboration
      </li>
      <li>
        Namecheap: Domain name registration
      </li>
      <li>
        Route 53: Domain name resolution
      </li>
      <li>
        AWS Amplify: Frontend server hosting
      </li>
      <li>
        AWS EC2: Backend server hosting
      </li>
      <li>
        AWS RDS: Database hosting
      </li>
      <li>
        Postman: API Documentation
      </li>
      <li>
        Selenium/Jest: Testing Frameworks
      </li>
    </ul>
    <h2>
      Data sources:
    </h2>
    <ul>
      <li>
        <a
          href="http://api.weatherapi.com/v1/forecast.json"
        >
          Weather API
        </a>
      </li>
      <li>
        <a
          href="https://fusion.yelp.com"
        >
          Yelp Fusion
        </a>
      </li>
      <li>
        <a
          href="https://developers.google.com/maps/documentation/geocoding/overview"
        >
          GeoCoding API
        </a>
      </li>
      <li>
        <a
          href="https://www.geoapify.com/geocoding-api"
        >
          GeoApify
        </a>
      </li>
    </ul>
    <p>
      Each of these sites provides a RESTful API. For Phase 1, we first analyzed the documentation to find the appropriate endpoints. Then, we scraped the data using a curl request and filtered the data by hand to find the appropriate information. Finally, we added the data to a JSON file so that it could be re-used across different pages.
    </p>
    <br />
  </div>
</div>
`;

exports[`Init App 1`] = `
Array [
  <nav
    className="navbar"
  >
    <ul
      className="nav-list"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
        >
          StormShelters
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/about"
        >
          About
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/cities"
        >
          Cities
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/shelter"
        >
          Shelter & Relief
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/pharmacies"
        >
          Pharmacies
        </a>
      </li>
    </ul>
  </nav>,
  <div
    className="home-container"
  >
    <div
      className="bg"
    >
      <div
        className="text-container"
      >
        <h1>
          Prepare for the Storms.
        </h1>
        <button
          className="button"
          id="scrollButton"
        >
          Find Resources
        </button>
      </div>
    </div>
    <div
      className="shelter-card-container"
      id="shelterCardContainer"
    >
      <a
        href="/cities"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <div
          className="shelter-card"
        >
          <img
            alt="Food and Pharmacies Image"
            class="shelter-image-container"
            src="https://ane4bf-datap1.s3-eu-west-1.amazonaws.com/wmocms/s3fs-public/styles/featured_media_detail/public/23613146140_6edfea579f_k.jpg?m0oNCnmjJAswk0CJrWZ.iW.ge6XbrfR1&itok=YBQjaq77"
          />
          <div
            className="shelter-card-body"
          >
            <h3
              className="shelter-card-title"
            >
              Cities
            </h3>
            <p>
              View the weather conditions of your city to see if a storm is in the forecast.
            </p>
          </div>
          <div
            className="button-container"
          />
        </div>
      </a>
      <a
        href="/shelter"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <div
          className="shelter-card"
        >
          <img
            class="shelter-image-container"
            src="https://i.cbc.ca/1.5499955.1584413818!/fileImage/httpImage/image.jpg_gen/derivatives/16x9_780/welcome-inn.jpg"
          />
          <div
            className="shelter-card-body"
          >
            <h3
              className="shelter-card-title"
            >
              Shelters & Food
            </h3>
            <p>
              View the areas to take refuge in Harris County and the highest-rated locations where you can stock up for the next storm or replenish your supplies while you recover. Filter by address to find the most convenient location for you.
            </p>
          </div>
          <div
            className="button-container"
          />
        </div>
      </a>
      <a
        href="/pharmacies"
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        <div
          className="shelter-card"
        >
          <img
            alt="Pharmacy Image"
            class="shelter-image-container"
            src="https://medicopharmacyrx.com/wp-content/uploads/2021/07/types-of-pharmacy.jpg"
          />
          <div
            className="shelter-card-body"
          >
            <h3
              className="shelter-card-title"
            >
              Pharmacies
            </h3>
            <p>
              Find where you can purchase essential medical supplies closest to you.
            </p>
          </div>
        </div>
      </a>
    </div>
  </div>,
]
`;

exports[`Init HomeCard 1`] = `
<div
  className="home-container"
>
  <div
    className="bg"
  >
    <div
      className="text-container"
    >
      <h1>
        Prepare for the Storms.
      </h1>
      <button
        className="button"
        id="scrollButton"
      >
        Find Resources
      </button>
    </div>
  </div>
  <div
    className="shelter-card-container"
    id="shelterCardContainer"
  >
    <a
      href="/cities"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="shelter-card"
      >
        <img
          alt="Food and Pharmacies Image"
          class="shelter-image-container"
          src="https://ane4bf-datap1.s3-eu-west-1.amazonaws.com/wmocms/s3fs-public/styles/featured_media_detail/public/23613146140_6edfea579f_k.jpg?m0oNCnmjJAswk0CJrWZ.iW.ge6XbrfR1&itok=YBQjaq77"
        />
        <div
          className="shelter-card-body"
        >
          <h3
            className="shelter-card-title"
          >
            Cities
          </h3>
          <p>
            View the weather conditions of your city to see if a storm is in the forecast.
          </p>
        </div>
        <div
          className="button-container"
        />
      </div>
    </a>
    <a
      href="/shelter"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="shelter-card"
      >
        <img
          class="shelter-image-container"
          src="https://i.cbc.ca/1.5499955.1584413818!/fileImage/httpImage/image.jpg_gen/derivatives/16x9_780/welcome-inn.jpg"
        />
        <div
          className="shelter-card-body"
        >
          <h3
            className="shelter-card-title"
          >
            Shelters & Food
          </h3>
          <p>
            View the areas to take refuge in Harris County and the highest-rated locations where you can stock up for the next storm or replenish your supplies while you recover. Filter by address to find the most convenient location for you.
          </p>
        </div>
        <div
          className="button-container"
        />
      </div>
    </a>
    <a
      href="/pharmacies"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <div
        className="shelter-card"
      >
        <img
          alt="Pharmacy Image"
          class="shelter-image-container"
          src="https://medicopharmacyrx.com/wp-content/uploads/2021/07/types-of-pharmacy.jpg"
        />
        <div
          className="shelter-card-body"
        >
          <h3
            className="shelter-card-title"
          >
            Pharmacies
          </h3>
          <p>
            Find where you can purchase essential medical supplies closest to you.
          </p>
        </div>
      </div>
    </a>
  </div>
</div>
`;

exports[`Init NavBar 1`] = `
<nav
  className="navbar"
>
  <ul
    className="nav-list"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/"
      >
        StormShelters
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/about"
      >
        About
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/cities"
      >
        Cities
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/shelter"
      >
        Shelter & Relief
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/pharmacies"
      >
        Pharmacies
      </a>
    </li>
  </ul>
</nav>
`;
